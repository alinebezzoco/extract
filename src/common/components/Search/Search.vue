<template>
  <label for="id">{{ label }}</label>
  <input type="text" :value="value" :name="name" :id="id" :placeholder="placeholder" data-testid="search-filter"
    @input="$emit('input', $event.target.value)" />
</template>

<script>
export default {
  name: "SearchComponent",
  props: {
    value: String,
    id: String,
    name: String,
    placeholder: String,
    label: String
  },
};
</script>