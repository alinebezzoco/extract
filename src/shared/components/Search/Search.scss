.search { 
    input { 
        height: 64px;
        background: #F0F4F8;
        border-radius: 16px;
        text-align: left;
        font-size: 16px;
        width: 600px;
        margin: 0 auto;
        display: block;
        padding: 0 20px;
        border: none;
        @media(max-width: 780px) { 
            width: 300px;
        }
    }
}