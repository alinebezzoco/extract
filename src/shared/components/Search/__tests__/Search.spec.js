import { mount } from "@vue/test-utils";
import { render, cleanup } from "@testing-library/vue";
import "@testing-library/jest-dom";
import Search from "../Search.vue";

const PROPS = {
  label: "Busca:",
  name: "input-search",
  id: "input-search",
  placeholder: "Digite...",
};

afterEach(cleanup);

describe("Search.vue", () => {
  it("should have render with default classes", () => {
    const container = mount(Search, {
      propsData: { 
        ...PROPS, 
        value: "Luis InÃ¡cio"
      },

    });
    expect(container).toMatchSnapshot();
  });

  it("should have focus input", () => {
    const { getByTestId } = render(Search, {
      propsData: PROPS,
    });
    getByTestId("search-filter").focus();
    expect(getByTestId("search-filter")).toHaveFocus();
  });

  it("should render empty value", () => {
    const container = mount(Search, {
      propsData: {
        ...PROPS,
        value: "",
      },
    });

    expect(container.value).toBeUndefined();
  });
});
