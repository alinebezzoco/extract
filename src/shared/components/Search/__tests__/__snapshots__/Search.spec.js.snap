// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search.vue should have render with default classes 1`] = `
VueWrapper {
  "__app": Object {
    "_component": Object {
      "name": "VTU_ROOT",
      "render": [Function],
    },
    "_container": <div
      data-v-app=""
    >
      
      <label
        for="id"
      >
        Busca:
      </label>
      <input
        data-testid="search-filter"
        id="input-search"
        name="input-search"
        placeholder="Digite..."
        type="text"
      />
      
    </div>,
    "_context": Object {
      "app": [Circular],
      "components": Object {
        "transition": Object {
          "name": "transition",
        },
        "transition-group": Object {
          "name": "transition-group",
        },
      },
      "config": Object {
        "compilerOptions": Object {},
        "errorHandler": undefined,
        "globalProperties": Object {},
        "isNativeTag": [Function],
        "optionMergeStrategies": Object {},
        "performance": false,
        "warnHandler": undefined,
      },
      "directives": Object {},
      "emitsCache": WeakMap {},
      "mixins": Array [
        Object {
          "beforeCreate": [Function],
        },
      ],
      "optionsCache": WeakMap {},
      "propsCache": WeakMap {},
      "provides": Object {},
      "reload": [Function],
    },
    "_instance": Object {
      "a": null,
      "accessCache": Object {},
      "appContext": Object {
        "app": [Circular],
        "components": Object {
          "transition": Object {
            "name": "transition",
          },
          "transition-group": Object {
            "name": "transition-group",
          },
        },
        "config": Object {
          "compilerOptions": Object {},
          "errorHandler": undefined,
          "globalProperties": Object {},
          "isNativeTag": [Function],
          "optionMergeStrategies": Object {},
          "performance": false,
          "warnHandler": undefined,
        },
        "directives": Object {},
        "emitsCache": WeakMap {},
        "mixins": Array [
          Object {
            "beforeCreate": [Function],
          },
        ],
        "optionsCache": WeakMap {},
        "propsCache": WeakMap {},
        "provides": Object {},
        "reload": [Function],
      },
      "asyncDep": null,
      "asyncResolved": false,
      "attrs": Object {},
      "bc": null,
      "bm": null,
      "bu": null,
      "bum": null,
      "c": null,
      "components": null,
      "ctx": Object {},
      "da": null,
      "data": Object {},
      "directives": null,
      "ec": null,
      "effect": ReactiveEffect {
        "active": true,
        "allowRecurse": true,
        "deps": Array [
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
          Set {
            [Circular],
          },
        ],
        "fn": [Function],
        "onTrack": undefined,
        "onTrigger": undefined,
        "parent": undefined,
        "scheduler": [Function],
      },
      "emit": [Function],
      "emitsOptions": null,
      "emitted": null,
      "exposeProxy": null,
      "exposed": null,
      "inheritAttrs": undefined,
      "isDeactivated": false,
      "isMounted": true,
      "isUnmounted": false,
      "m": null,
      "next": null,
      "parent": null,
      "props": Object {},
      "propsDefaults": Object {},
      "propsOptions": Array [
        Object {},
        Array [],
      ],
      "provides": Object {},
      "proxy": Object {},
      "refs": Object {
        "VTU_COMPONENT": Object {
          "hasOwnProperty": [Function],
          "id": "input-search",
          "label": "Busca:",
          "name": "input-search",
          "placeholder": "Digite...",
          "value": "Luis In치cio",
        },
      },
      "render": [Function],
      "renderCache": Array [],
      "root": [Circular],
      "rtc": null,
      "rtg": null,
      "scope": EffectScope {
        "active": true,
        "cleanups": Array [],
        "effects": Array [
          ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
        ],
      },
      "setupContext": null,
      "setupState": Object {},
      "slots": Object {},
      "sp": null,
      "subTree": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": null,
        "children": null,
        "component": Object {
          "a": null,
          "accessCache": Object {
            "@@__IMMUTABLE_ITERABLE__@@": 0,
            "@@__IMMUTABLE_RECORD__@@": 0,
            "_isMockFunction": 0,
            "constructor": 0,
            "hasOwnProperty": 4,
            "id": 3,
            "isVueInstance": 0,
            "label": 3,
            "name": 3,
            "placeholder": 3,
            "toJSON": 0,
            "value": 3,
            Symbol(Symbol.toStringTag): 0,
          },
          "appContext": Object {
            "app": [Circular],
            "components": Object {
              "transition": Object {
                "name": "transition",
              },
              "transition-group": Object {
                "name": "transition-group",
              },
            },
            "config": Object {
              "compilerOptions": Object {},
              "errorHandler": undefined,
              "globalProperties": Object {},
              "isNativeTag": [Function],
              "optionMergeStrategies": Object {},
              "performance": false,
              "warnHandler": undefined,
            },
            "directives": Object {},
            "emitsCache": WeakMap {},
            "mixins": Array [
              Object {
                "beforeCreate": [Function],
              },
            ],
            "optionsCache": WeakMap {},
            "propsCache": WeakMap {},
            "provides": Object {},
            "reload": [Function],
          },
          "asyncDep": null,
          "asyncResolved": false,
          "attrs": Object {},
          "bc": null,
          "bm": null,
          "bu": null,
          "bum": null,
          "c": null,
          "components": Object {},
          "ctx": Object {
            "hasOwnProperty": [Function],
            "id": "input-search",
            "label": "Busca:",
            "name": "input-search",
            "placeholder": "Digite...",
            "value": "Luis In치cio",
          },
          "da": null,
          "data": Object {},
          "directives": null,
          "ec": null,
          "effect": ReactiveEffect {
            "active": true,
            "allowRecurse": true,
            "deps": Array [
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
              Set {
                [Circular],
              },
            ],
            "fn": [Function],
            "onTrack": undefined,
            "onTrigger": undefined,
            "parent": undefined,
            "scheduler": [Function],
          },
          "emit": [Function],
          "emitsOptions": null,
          "emitted": null,
          "exposeProxy": null,
          "exposed": null,
          "inheritAttrs": undefined,
          "isDeactivated": false,
          "isMounted": true,
          "isUnmounted": false,
          "m": null,
          "next": null,
          "parent": [Circular],
          "props": Object {
            "id": "input-search",
            "label": "Busca:",
            "name": "input-search",
            "placeholder": "Digite...",
            "value": "Luis In치cio",
          },
          "propsDefaults": Object {},
          "propsOptions": Array [
            Object {
              "id": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "label": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "name": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "placeholder": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
              "value": Object {
                "0": false,
                "1": true,
                "type": [Function],
              },
            },
            Array [],
          ],
          "provides": Object {},
          "proxy": Object {
            "hasOwnProperty": [Function],
            "id": "input-search",
            "label": "Busca:",
            "name": "input-search",
            "placeholder": "Digite...",
            "value": "Luis In치cio",
          },
          "refs": Object {},
          "render": [Function],
          "renderCache": Array [
            [Function],
          ],
          "root": [Circular],
          "rtc": null,
          "rtg": null,
          "scope": EffectScope {
            "active": true,
            "cleanups": Array [],
            "effects": Array [
              ReactiveEffect {
                "active": true,
                "allowRecurse": true,
                "deps": Array [
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                  Set {
                    [Circular],
                  },
                ],
                "fn": [Function],
                "onTrack": undefined,
                "onTrigger": undefined,
                "parent": undefined,
                "scheduler": [Function],
              },
            ],
          },
          "setupContext": null,
          "setupState": Object {},
          "slots": Object {},
          "sp": null,
          "subTree": Object {
            "__v_isVNode": true,
            "__v_skip": true,
            "anchor": ,
            "appContext": null,
            "children": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Busca:",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <label
                  for="id"
                >
                  Busca:
                </label>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "for": "id",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "value",
                  "name",
                  "id",
                  "placeholder",
                ],
                "el": <input
                  data-testid="search-filter"
                  id="input-search"
                  name="input-search"
                  placeholder="Digite..."
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": Object {
                  "data-testid": "search-filter",
                  "id": "input-search",
                  "name": "input-search",
                  "onInput": [Function],
                  "placeholder": "Digite...",
                  "type": "text",
                  "value": "Luis In치cio",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "component": null,
            "dirs": null,
            "dynamicChildren": Array [
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": "Busca:",
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": null,
                "el": <label
                  for="id"
                >
                  Busca:
                </label>,
                "key": null,
                "patchFlag": 1,
                "props": Object {
                  "for": "id",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 9,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "label",
              },
              Object {
                "__v_isVNode": true,
                "__v_skip": true,
                "anchor": null,
                "appContext": null,
                "children": null,
                "component": null,
                "dirs": null,
                "dynamicChildren": null,
                "dynamicProps": Array [
                  "value",
                  "name",
                  "id",
                  "placeholder",
                ],
                "el": <input
                  data-testid="search-filter"
                  id="input-search"
                  name="input-search"
                  placeholder="Digite..."
                  type="text"
                />,
                "key": null,
                "patchFlag": 40,
                "props": Object {
                  "data-testid": "search-filter",
                  "id": "input-search",
                  "name": "input-search",
                  "onInput": [Function],
                  "placeholder": "Digite...",
                  "type": "text",
                  "value": "Luis In치cio",
                },
                "ref": null,
                "scopeId": null,
                "shapeFlag": 1,
                "slotScopeIds": null,
                "ssContent": null,
                "ssFallback": null,
                "staticCount": 0,
                "suspense": null,
                "target": null,
                "targetAnchor": null,
                "transition": null,
                "type": "input",
              },
            ],
            "dynamicProps": null,
            "el": ,
            "key": null,
            "patchFlag": 64,
            "props": null,
            "ref": null,
            "scopeId": null,
            "shapeFlag": 16,
            "slotScopeIds": null,
            "ssContent": null,
            "ssFallback": null,
            "staticCount": 0,
            "suspense": null,
            "target": null,
            "targetAnchor": null,
            "transition": null,
            "type": Symbol(Fragment),
          },
          "suspense": null,
          "suspenseId": 0,
          "type": Object {
            "components": Object {},
            "name": "SearchComponent",
            "props": Object {
              "id": [Function],
              "label": [Function],
              "name": [Function],
              "placeholder": [Function],
              "value": [Function],
            },
            "propsData": Object {
              "id": "input-search",
              "label": "Busca:",
              "name": "input-search",
              "placeholder": "Digite...",
              "value": "Luis In치cio",
            },
            "render": [Function],
          },
          "u": null,
          "uid": 1,
          "um": null,
          "update": [Function],
          "vnode": [Circular],
          "withProxy": null,
        },
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": Object {
          "id": "input-search",
          "label": "Busca:",
          "name": "input-search",
          "placeholder": "Digite...",
          "ref": "VTU_COMPONENT",
          "value": "Luis In치cio",
        },
        "ref": Object {
          "f": false,
          "i": [Circular],
          "k": undefined,
          "r": "VTU_COMPONENT",
        },
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "components": Object {},
          "name": "SearchComponent",
          "props": Object {
            "id": [Function],
            "label": [Function],
            "name": [Function],
            "placeholder": [Function],
            "value": [Function],
          },
          "propsData": Object {
            "id": "input-search",
            "label": "Busca:",
            "name": "input-search",
            "placeholder": "Digite...",
            "value": "Luis In치cio",
          },
          "render": [Function],
        },
      },
      "suspense": null,
      "suspenseId": 0,
      "type": Object {
        "name": "VTU_ROOT",
        "render": [Function],
      },
      "u": null,
      "uid": 0,
      "um": null,
      "update": [Function],
      "vnode": Object {
        "__v_isVNode": true,
        "__v_skip": true,
        "anchor": null,
        "appContext": Object {
          "app": [Circular],
          "components": Object {
            "transition": Object {
              "name": "transition",
            },
            "transition-group": Object {
              "name": "transition-group",
            },
          },
          "config": Object {
            "compilerOptions": Object {},
            "errorHandler": undefined,
            "globalProperties": Object {},
            "isNativeTag": [Function],
            "optionMergeStrategies": Object {},
            "performance": false,
            "warnHandler": undefined,
          },
          "directives": Object {},
          "emitsCache": WeakMap {},
          "mixins": Array [
            Object {
              "beforeCreate": [Function],
            },
          ],
          "optionsCache": WeakMap {},
          "propsCache": WeakMap {},
          "provides": Object {},
          "reload": [Function],
        },
        "children": null,
        "component": [Circular],
        "dirs": null,
        "dynamicChildren": null,
        "dynamicProps": null,
        "el": ,
        "key": null,
        "patchFlag": 0,
        "props": null,
        "ref": null,
        "scopeId": null,
        "shapeFlag": 4,
        "slotScopeIds": null,
        "ssContent": null,
        "ssFallback": null,
        "staticCount": 0,
        "suspense": null,
        "target": null,
        "targetAnchor": null,
        "transition": null,
        "type": Object {
          "name": "VTU_ROOT",
          "render": [Function],
        },
      },
      "withProxy": null,
    },
    "_props": null,
    "_uid": 0,
    "component": [Function],
    "config": Object {
      "compilerOptions": Object {},
      "errorHandler": undefined,
      "globalProperties": Object {},
      "isNativeTag": [Function],
      "optionMergeStrategies": Object {},
      "performance": false,
      "warnHandler": undefined,
    },
    "directive": [Function],
    "mixin": [Function],
    "mount": [Function],
    "provide": [Function],
    "unmount": [Function],
    "use": [Function],
    "version": "3.2.37",
  },
  "__setProps": [Function],
  "componentVM": Object {
    "hasOwnProperty": [Function],
    "id": "input-search",
    "label": "Busca:",
    "name": "input-search",
    "placeholder": "Digite...",
    "value": "Luis In치cio",
  },
  "isDisabled": [Function],
  "rootVM": Object {},
  "wrapperElement": ,
}
`;
